filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsPathBuilderFactory.scala

```diff
@@ -36,96 +27,46 @@ class DrsPathBuilderFactory(globalConfig: Config, instanceConfig: Config, single
-              if storageException.getMessage == "Bucket is requester pays bucket but no user project provided." =>
-              IO(
-                storage
-                  .get(bucketName, objectName, BlobGetOption.userProject(requesterPaysProjectId))
-                  .reader(Blob.BlobSourceOption.userProject(requesterPaysProjectId))
-              )
-            case _ => IO.raiseError(throwable)
-          }
-      }
-    } yield readChannel
-  }
-
-
-  private def drsReadInterpreter(options: WorkflowOptions, requesterPaysProjectIdOption: Option[String])
-                                (gsUri: Option[String], googleServiceAccount: Option[SADataObject])
-  : IO[ReadableByteChannel] = {
-    val readScopes = List(StorageScopes.DEVSTORAGE_READ_ONLY)
-    val credentialsIo = googleServiceAccount match {
-      case Some(googleSA) =>
-        IO(
-          UserServiceAccountMode("martha_service_account").credentials(
-            Map(GoogleAuthMode.UserServiceAccountKey -> googleSA.data.noSpaces),
-            readScopes,
-          )
-        )
-      case None =>
-        IO(googleAuthMode.credentials(options.get(_).get, readScopes))
-    }
-
-    for {
-      credentials <- credentialsIo
-      //Currently, Martha only supports resolving DRS paths to GCS paths
-      gsUri <- IO.fromEither(gsUri.toRight(MarthaResponseMissingKeyException("gsUri")))
-      readableByteChannel <- gcsInputStream(gsUri, credentials, requesterPaysProjectIdOption)
-    } yield readableByteChannel
-  }
-
-
   override def withOptions(options: WorkflowOptions)(implicit as: ActorSystem, ec: ExecutionContext): Future[PathBuilder] = {
-    val marthaScopes = List(
-      // Profile and Email scopes are requirements for interacting with Martha
-      Oauth2Scopes.USERINFO_EMAIL,
-      Oauth2Scopes.USERINFO_PROFILE
-    )
-    val authCredentials = googleAuthMode.credentials(options.get(_).get, marthaScopes)
-
-    // Unlike PAPI we're not going to fall back to a "default" project from the backend config.
-    // ONLY use the project id from the User Service Account for requester pays
-    val requesterPaysProjectIdOption = options.get("google_project").toOption
-
-    /*
-    `override_preresolve_for_test` is a workflow option to override the default `martha.preresolve` specified in the
-    global config. This is only used for testing purposes.
-     */
-    val preResolve: Boolean =
-      options
-        .getBoolean("override_preresolve_for_test")
-        .toOption
-        .getOrElse(
-          singletonConfig
-            .config
-            .getBoolean("martha.preresolve")
-        )
+    Future {
+      val marthaScopes = List(
+        // Profile and Email scopes are requirements for interacting with Martha
+        Oauth2Scopes.USERINFO_EMAIL,
+        Oauth2Scopes.USERINFO_PROFILE
+      )
+      val authCredentials = googleAuthMode.credentials(options.get(_).get, marthaScopes)
```

Comment:
This validates credentials so it could throw, which now will result in a failed future instead of an exception being thrown right away. However, The only significant (non-test) path that I found was from `PathBuilderFactory.instantiatePathBuilders` which itself seems to be mostly trace up to `lazy` vals which, I believe, effectively defers evaluation until when the `Future` is being resolved anyway. Therefore, I don't think there is any significant change in the path that validation errors will propagate up the stack. 👍